### DISCUSSION TITLE
How can i access TestResult object from code

### DISCUSSION THREAD

**Discussion by Rjakubiec** [allure-framework/allure2#1923] [General Discussion]
Hello,

I have a simple and possibly trivial question: is it possible to access the TestResult object from code? I tried using Allure.getLifecycle(), but I couldn't find any way to retrieve the data saved in the TestResult object.

**Comment by baev** [comment]
You can use `Allure.getLifecycle().updateTestCase(Consumer<TestResult> consumer)` method to modify the current test result. If case you need to extract it to separate variable, you can set the reference in consumer:

```java
TestResult result = null;
Allure.getLifecycle().updateTestCase(testResult -> { result = testResult });
```

**Reply by Rjakubiec** [reply]
Great, it works very well.
 I have only one additional question: if I use that solution in the test body, it works great.

    @Test
    @TmsLink("")
    public void test() throws NoSuchFieldException {
        List<Link> links = new ArrayList<>();
        links.add(new Link().setName("test").setType("tms"));
        Allure.getLifecycle().updateTestCase(t->t.setLinks(links));
        List<TestResult> result = new ArrayList<>();
        Allure.getLifecycle().updateTestCase(result::add);
        result.get(0).getLinks();
   }
// The 'result' variable contains the set link
 

 However, if I use that solution in the @AfterMethod, the TestResult is empty. 

    @Test
    @TmsLink("")
    public void test() throws NoSuchFieldException {
        List<Link> links = new ArrayList<>();
        links.add(new Link().setName("test").setType("tms"));
        Allure.getLifecycle().updateTestCase(t->t.setLinks(links));

    }

    @AfterMethod(alwaysRun = true)
    public void afterTest(ITestResult testResult) {

        List<TestResult> result = new ArrayList<>();
        Allure.getLifecycle().updateTestCase(result::add);
        result.get(0).getLinks();
    }
// The 'result' variable is empty 


I am using TestNG.
